BML_TUX_Project/Packages/bmlTUX/Scripts/Settings/FileLocationSettings.cs

```diff
@@ -3,48 +3,52 @@
 using UnityEngine;
 
 namespace bmlTUX {
-	
-	public static class FileLocationSettings  {
-
-		public const string TuxFolderName = "bmlTux";
-		public const string ScriptingTemplatesFolderName =	"ScriptingTemplates";
-		public const string PackagePath = "Packages/com.biomotionlab.tux";
-		
-
-		public const string LastSessionSaveFileName = "lastSession.json";
-		public const string SessionLogFileName = "sessionLog.txt";
-		public const  string DebugFileName = "debugSaveFile";
-		public const string tuxFolderName = "bmlTUX_Data";
-		
-		
-		public static string TemplatePath => Path.Combine(TuxFolderPath, ScriptingTemplatesFolderName);
-
-		public static string TuxFolderPath {
-			get {
-				string tuxPath;
-				string tuxPackagePath = "";
-				string tuxAssetPath = Path.Combine(Application.dataPath, TuxFolderName);
-				tuxPath = tuxAssetPath;
-				if (!Directory.Exists(tuxPath)) {
-					tuxPackagePath = Path.GetFullPath(PackagePath);
-					tuxPath = tuxPackagePath;
-				}
-				
-				if (!Directory.Exists(tuxPath)) {
-					throw new DirectoryNotFoundException($"Can't find bmlTUX directory. Tried: {tuxAssetPath}, and {tuxPackagePath}");
-				}
-
-				return tuxPath;
-			}
-		}
-
-		
-		
-		public static string BaseTuxDocumentsFolderPath => Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments), tuxFolderName);
-		public static string SessionFolder => BaseTuxDocumentsFolderPath;
-		public static string DebugFolder => BaseTuxDocumentsFolderPath;
-		
-		public static string LastSessionSaveFilePath => Path.Combine(SessionFolder, LastSessionSaveFileName);
-		public static string SessionLogFilePath => Path.Combine(SessionFolder, SessionLogFileName);
-	}
+  [Serializable]
+	public class FileLocationSettings  {
```

Comment:
I feel like this class should be a scriptable object. There could be a default version of it with the correct values in the package folder, that is automatically referenced by the design file. The user can create their own version of the scriptable object in their own project and override it if needed (drag theirs in to replace the default reference).

Reply:
I agree this feels right, but this is more difficult than I expected it to be without breaking anything, because any existing Experiment Settings files will become outdated and have to be updated with a reference to the default version. I am unsure how to progress here.

I considered using another script to check for this missing reference in any existing Design / Experiment Settings asset in the project, something like another AssetPostProcess Script filtering the AssetDatabase, but that would trigger on every AssetPostProcess, not just after updating the package. Also that just seems like a hack to avoid a breaking change.

Alternatively, ExperimentDesignFile2 has a Validator that could throw a MissingReference, but I think user will be confused at first where to find the default version if its located in the package, or might not get the idea of creating a new ScriptableObject immediatly.

As far as I understand, other references to Scriptable Objects are setup during initial experiment creation, when utilizing ScriptHelperEditorWindow. Right now I cant think of a way to update existing Designs with a default FileLocationSettings that feels satisfactory. Do you have a good idea?

Reply:
You can set a default serialized field's asset reference (Prefab or scriptable object) in the _**Script's**_ inspector. 

So :
1. Create the new file settings default scriptable object. Put it in a good spot in the BMLtux package. (not in the assets folder)
2. Add the serialized field to the experiment design file script. 
3. Find the experiment design file SCRIPT (not an instance of it) in the project view. Click on it and take a look at its inspector. Drag the new file settings SO into the field. 
4. This will ONLY apply on the very first serialization of this field. So in your local project, you'll see nothing happen because you already serialized the field in between steps 2 and 3.
5. But you can test if it worked by creating a new design file and double check the reference is there. Now you can hook it up in your project's design file manually.

![image](https://github.com/BioMotionLab/TUX/assets/43276915/7e20473d-d2c2-4a53-a28f-5b071efd4363)

Let me know you have any questions, this is a pretty obscure Unity feature not too many people know about.

Reply:
Thanks for your great advice! I did not know this feature existed.

Testing this, I wasnt able to validate 6. 
It's possible I am not testing it correctly though. I updated a test project by copying a local version of the package and reimported everything. Not sure if it'll behave differently if it's getting an update from the package manager.

Reply:
Changed in: 5a5212136137a1548ee3cc3f043330f7d107c73b

Reply:
My Project is on 2021.3.23f1